### DISCUSSION TITLE
How can I change channel permissions for role/user?

### DISCUSSION THREAD

**Discussion by at-elcapitan** [discord-jda/JDA#2530] [Questions and Help] [answered]
Greetings. When creating an administrative bot, the task arose to manipulate the privileges for individual users/roles for a particular channel. Are there any options to implement this?

**Comment by MinnDevelopment** [comment, marked as answer]
With [upserPermissionOverride](https://docs.jda.wiki/net/dv8tion/jda/api/entities/channel/attribute/IPermissionContainer.html#upsertPermissionOverride(net.dv8tion.jda.api.entities.IPermissionHolder)) you can change permission overrides by using [grant](https://docs.jda.wiki/net/dv8tion/jda/api/requests/restaction/PermissionOverrideAction.html#grant(net.dv8tion.jda.api.Permission...)) and [deny](https://docs.jda.wiki/net/dv8tion/jda/api/requests/restaction/PermissionOverrideAction.html#deny(net.dv8tion.jda.api.Permission...)).
